newIDE/app/src/EventsFunctionsExtensionsLoader/CodeWriters/LocalEventsFunctionCodeWriter.js

```diff
@@ -16,7 +16,14 @@ export const makeLocalEventsFunctionCodeWriter = ({
   onWriteFile,
 }: EventsFunctionCodeWriterCallbacks): EventsFunctionCodeWriter => {
   // The generated code for extensions will be stored in a temporary directory
-  const outputDir = os.tmpdir() + '/GDGeneratedEventsFunctions';
+  var outputDirTemp ;
```

Comment:
Never use var, always `const`.

Reply:
Again I was forced to do like so. And this variable is temporary. It's because there was issues with variable scopes.

Reply:
You can always abstract the call to `os.userInfo()` and the try/catch in a function ;)

Reply:
Can we define a blank const and set the value later ?

Reply:
No - the goal with `const` is to never change your variable. Here the code is too complicated because the try catch is forcing you to use var/let.
Instead, make a function to abstract the call to userInfo(). You can call this function `getUsername`. In this function, you get first the username and return it. If there is an error, you return an empty string. 
Simple and clean :)

Reply:
Well I wasn't aware we can do that. I modified it but not the way you wanted to so I guess I have to start again ?

Reply:
I know it's frustrating :) But yes you should really follow what I recommend here (make a tiny function `getUsername`, it's only a few lines). 
It's important to keep a code as concise as possible.

Reply:
i'm sorry but... I don't know what is an "abstract"